lib/3scale/backend/transactor.rb

```diff
@@ -61,6 +61,8 @@ def utilization(service_id, application_id)
 
       def validate(oauth, provider_key, report_usage, params, request_info)
         service = Service.load_with_provider_key!(params[:service_id], provider_key)
+        oidc_service = !oauth && service.backend_version == 'oauth'.freeze
```

Comment:
Related to my other comment. I think the distinction between oidc and oauth is confusing.
I think we need a method that tells us if we need to apply the oauth rules (oauth endpoint or backend_version == 'oauth') or something similar.

Reply:
Related to https://github.com/3scale/apisonator/pull/280#discussion_r644708714

Reply:
Well, this is that method except it is not extracted out. The long term change would be to drop completely oauth_auth*.xml and the OAuth validators and treat everything in authorize and authrep as `user_key` and`app_id` (OIDC) on one hand, which don't need to check any app_keys, and `app_id` _with_ `app_key` on the other hand, which does require it.

The actual distinction may be made by checking the `backend_version` of the service or by deciding whether the lack of an `app_key` parameter is significative (ie. if the client, which is trusted, does not specify an `app_key` by omitting the parameter, then it can be interpreted as _not wanting_ the check on app keys).